File: Data Structures & Algorithms/buy-and-sell-crypto/submission-8.js
class Solution {
    /**
     * @param {number[]} prices
     * @return {number}
     */
    maxProfit(prices) {
        let buy = Infinity;
        let profit = 0;

        for(let price of prices){
           if(buy < price){
            profit = Math.max(profit, price - buy);
           }else{
            buy  = price;
           }
        }
        return profit;

    }
}

File: Data Structures & Algorithms/max-water-container/submission-8.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        
        l =  0
        r = len(heights) - 1
        maxArea = 0
        while l < r:
            h = min(heights[l], heights[r])
            w = r - l

            maxArea = max(h*w, maxArea)

            if heights[l] >= heights[r]:
                r -= 1
            else:
                l += 1
        return maxArea
